Fix firmware installation

When installing modules which have their firmware only in firmware/
(no subdirectory), the build fails with:

$ make INSTALL_MOD_PATH=/home/rtp/devel/kernel/BUILD/kernel-x86_64/temp-root KERNELRELEASE=2.6.38.2-desktop-2.mga modules_install 
  INSTALL  drivers/usb/serial/usbserial.ko
  INSTALL  drivers/usb/serial/whiteheat.ko
make[1]: *** No rule to make target `/home/rtp/devel/kernel/BUILD/kernel-x86_64/temp-root/lib/firmware/./', needed by `/home/rtp/devel/kernel/BUILD/kernel-x86_64/temp-root/lib/firmware/whiteheat_loader.fw'.  Stop.
make: *** [_modinst_post] Error 2

If one is building a kernel with a module like 3c359, which has its firmware 
in firmware/3com/, the mkdir -p firmware/3com/ will create the firmware/ 
directory, allowing to install firmwares in firmware/, masking the error.

I noticed it while building with make 3.82, so this problem may be related
to that.

Signed-off-by: Arnaud Patard <arnaud.patard@rtp-net.org>
Index: linux-2.6.38/scripts/Makefile.fwinst
===================================================================
--- linux-2.6.38.orig/scripts/Makefile.fwinst
+++ linux-2.6.38/scripts/Makefile.fwinst
@@ -27,7 +27,7 @@ endif
 installed-mod-fw := $(addprefix $(INSTALL_FW_PATH)/,$(mod-fw))
 
 installed-fw := $(addprefix $(INSTALL_FW_PATH)/,$(fw-shipped-all))
-installed-fw-dirs := $(sort $(dir $(installed-fw))) $(INSTALL_FW_PATH)/.
+installed-fw-dirs := $(sort $(dir $(installed-fw))) $(INSTALL_FW_PATH)/./
 
 # Workaround for make < 3.81, where .SECONDEXPANSION doesn't work.
 PHONY += $(INSTALL_FW_PATH)/$$(%) install-all-dirs
